src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -98,25 +98,41 @@ void LinearIR::serialize(const std::string& xml, const std::string& bin) const {
     ov::pass::Serialize(xml, bin).run_on_model(tmp_model);
 }
 
-LinearIR::container LinearIR::deep_copy_range(LinearIR::container::const_iterator begin, LinearIR::container::const_iterator end) {
-    auto deep_clone_ports = [](std::vector<PortDescriptorPtr>& ports) {
-        for (auto& port : ports) { port = port->clone(); }
-    };
-    LinearIR::container result;
+LinearIR::container LinearIR::deep_copy_range(LinearIR::container::const_iterator begin,
+                                              LinearIR::container::const_iterator end,
+                                              std::unordered_map<ExpressionPtr, ExpressionPtr>& expression_map) const {
```

Comment:
Please note: more general deep copy solution has already implemented in #20242. We should decide whether we need to reuse the general approach in this PR (and merge this PR only after#20242) or not